templates/al2023/runtime/gpu/set-nvidia-clocks.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o xtrace
+
+if ! gpu-ami-util has-nvidia-devices; then
+  echo >&2 "no NVIDIA devices are present, nothing to do!"
+  exit 0
+fi
+
+# gpu boost clock
+sudo nvidia-smi -pm 1 # set persistence mode
+sudo nvidia-smi --auto-boost-default=0
```

Comment:
in the final else case below, we should not disable auto boost if the card is unknown to this script

if a new instance is launched, we don't want to hamper its performance in any form if we skip updating this script - let it fall back to the default behavior of the card

Reply:
@bryantbiggs do you mind if we address this in another PR?

Reply:
absolutely - as stated below, the two added comments here are great candidates for follow PRs if we think the changes are worth. But I think this turkey is cooked as is so - :shipit: